libcob/fileio.c

```diff
@@ -920,19 +920,29 @@ bdb_bt_compare (DB *db, const DBT *k1, const DBT *k2
 {
 	const unsigned char *col = (unsigned char *)DBT_GET_APP_DATA (k1);
 	COB_UNUSED (db);
-
+#ifdef USE_BDB_KEYDIFF /* flag passed with CPPFLAGS */
+	COB_MODULE_PTR->collating_sequence = col;
+	unsigned char *data1 = k1->data;
+	unsigned char *data2 = k2->data;
+	size_t size1 = (size_t)k1->size;
+	size_t size2 = (size_t)k2->size;
+	return cob_cmp_strings (data1, data2, size1, size2, col);
```

Comment:
Is there a reason for intermediate variables here?

Reply:
There is no reason. It was temporary and they were meant to be removed.

Reply:
you've removed those here, but left them in line 2031++ https://github.com/OCamlPro/gnucobol/pull/156/files#diff-16de7d2ecd4918cd696dff002aebea147395b0d09288c2e8e0310155dc943da1R2031-R2037